src/Subscription.ts

```diff
@@ -66,11 +67,25 @@ export class Subscription implements ISubscription {
       return;
     }
 
-    this.closed = true;
-
-    const { _unsubscribe, _subscriptions } = (<any> this);
+    let { _parent, _parents, _unsubscribe, _subscriptions } = (<any> this);
 
-    (<any> this)._subscriptions = null;
+    this.closed = true;
+    this._parent = null;
+    this._parents = null;
+    // null out _subscriptions first so any child subscriptions that attempt
+    // to remove themselves from this subscription will noop
+    this._subscriptions = null;
+
+    let index = -1, len = _parents && _parents.length || 0;
```

Comment:
IMO we should prefer multiline declarations for clarity--that's just my opinion tho.